content/en/docs/concepts/cluster-administration/api-priority-and-fairness.md

```diff
@@ -0,0 +1,125 @@
+flow schema's _matching precedence_ (which is a number).  A flow
+schema has rules that describe which requests match that schema.
+Matching takes into account (a) the requested operation&emdash;as
+identified by the HTTP request verb and URI&emdash;and (b) if the request
+arrived through a secured port then the authenticated user attributes
+otherwise just the UserAgent.  Among the schemas that match the
+request, one with the logically highest matching precedence wins.
+
+A flow schema also has a declaration of how to complete the _flow
+identifier_ for the request's flow.  A flow identifier is a pair of
+strings: the name of its flow schema plus one more string called the
+_flow distinguisher_.  A flow schema indicates whether the
+distinguisher is (1) the username or UserAgent of the user making the
+request, (2) the namespace (if any) of the object that the request
+seeks to access, or (3) the empty string.
+
+A flow schema also states the name of the priority level to use for
+requests that match.
+
+## Queuing
+
+An apiserver first classifies a request to a flow, and thus a
+corresponding priority level, and if the priority level is not exempt
+then the apiserver assigns the request to a queue using a technique
+called "shuffle sharding".  This technique starts with hashing the
+flow identifier, and then uses the hash value as a source of entropy
+to pseudo-randomly pick a small subset of the queues, and finally
+assigns the request to one of the shortest queues in that subset.  The
+priority level declares the size of the subset.
+
+After assigning a request to a priority level and a queue, the
+apiserver immediately begins handling the request if that would not
+violate the priority level's concurrency limit; otherwise the queue
+holds the request until it can be executed or is evicted.
+
+The apiserver applies two criteria to limit queue occupancy.  First,
+each non-exempt priority level has a queue length limit.  The
+apiserver will not add a request to a queue that is already at or
+exceeding its length limit; rather, the request is immediately
+rejected.  Second, each apiserver is configured with a limit on how
+long a request can wait in a queue.  If that limit expires before the
+request can be handled then the request is rejected.  When rejecting
+requests, the apiserver responds using HTTP status code 429 "Too Many
+Requests".
+
+## Configuration
+
+Configuration is mainly through API objects of kinds
+`PriorityLevelConfiguration` and `FlowSchema` in the
+`flowcontrol.apiserver.k8s.io` API group.
+
+The apiserver's aggregate concurrency limit is the sum of the two
+limits applied by the older simpler functionality: the limit on the
+number of mutating requests being handled at one time and the limit on
+the number of readonly requests being handled at a time.
+
+An apiserver's limit on how long a request can wait in a queue is one
+quarter of the time that an apiserver allows itself for handling a
+request.
```

Comment:
I'd almost like to see this section placed into the above section title "Flow schema" and include an example YAML snippet showing how the operator can configure rate limiting for a particular namespace or resource type